CMakeLists.txt

```diff
@@ -35,8 +35,26 @@ FetchContent_Declare(
 set(CMAKE_POLICY_DEFAULT_CMP0077 NEW) 
 
 set(TRIESTE_BUILD_SAMPLES OFF)
+
+if (REGOCPP_SPECULATIVE_LOAD_HARDENING)
+  if (NOT CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+    message(FATAL_ERROR "Speculative load hardening is only supported with Clang")
+  endif()
+endif()
+
+if (REGOCPP_SPECULATIVE_LOAD_HARDENING)
+  # Also build dependencies with speculative load hardening
+  # when the flag is enabled
+  set(CMAKE_CXX_FLAGS_SAVED "${CMAKE_CXX_FLAGS}")
```

Comment:
Why is this needed? Surely applying them using `target_compile_options` is sufficient?

Reply:
Would it not be necessary to invoke target_compile_options() on all the targets we transitively depend on?